.github/workflows/push.yml

```diff
@@ -17,12 +21,40 @@ jobs:
         if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
         run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
 
-      - name: Publish Containers
+      - name: Publish Containers to GHCR
         if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
         run: |
           GIT_VERSION=$(git describe --tags --always --match 'v*')
+          echo "GIT_VERSION=$GIT_VERSION" >> $GITHUB_ENV
 
           docker tag argo-tasks ghcr.io/linz/argo-tasks:latest
           docker tag argo-tasks ghcr.io/linz/argo-tasks:${GIT_VERSION}
 
           docker push --all-tags ghcr.io/linz/argo-tasks
+
+      - name: Configure AWS Credentials
+        if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ap-southeast-2
+          mask-aws-account-id: true
+          role-to-assume: ${{ secrets.AWS_CI_ROLE }}
+
+      - name: Login to Amazon ECR
+        if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Publish Containers to ECR
+        if: ${{(github.ref == 'refs/heads/master') && !(startsWith(github.event.head_commit.message, 'release:'))}}
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          REPOSITORY: eks
+          IMAGE_TAG: ${{ github.sha }}
+        run: |
+          docker pull ghcr.io/linz/argo-tasks:${{ env.GIT_VERSION }}
```

Comment:
can we not just use the previously built `argo-tasks` and tag it here, why are we pulling this container?